### DISCUSSION TITLE
Build failed

### DISCUSSION THREAD

**Discussion by har1s1m** [MystenLabs/sui#3377] [General]
Hey there

I just recently discovered this project and decided to try it out ;)

I'm trying to build using my remote virtual machine. I was following all steps provided in building from the source(official doc) and I got an error 

`
  Compiling prost-build v0.10.4
   Compiling openssl v0.10.40
   Compiling openssl-sys v0.9.74
   Compiling native-tls v0.2.10
   Compiling zstd-sys v1.6.3+zstd.1.5.2
error: failed to run custom build command for `prost-build v0.10.4`

Caused by:
  process didn't exit successfully: `/root/sui/target/release/build/prost-build-94a76a751467268e/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=/root/.cargo/registry/src/github.com-1ecc6299db9ec823/prost-build-0.10.4/third-party/protobuf/cmake
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: "cmake" "/root/.cargo/registry/src/github.com-1ecc6299db9ec823/prost-build-0.10.4/third-party/protobuf/cmake" "-Dprotobuf_BUILD_TESTS=OFF" "-DCMAKE_INSTALL_PREFIX=/root/sui/target/release/build/prost-build-604a95da7246c98e/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Debug"
  --
  -- 3.19.4.0
  -- Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /root/sui/target/release/build/prost-build-604a95da7246c98e/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Debug" "--parallel" "4"

  --- stderr
  Unknown argument --parallel
  Unknown argument 4
  Usage: cmake --build <dir> [options] [-- [native-options]]
  Options:
    <dir>          = Project binary directory to be built.
    --target <tgt> = Build <tgt> instead of default targets.
                     May only be specified once.
    --config <cfg> = For multi-configuration tools, choose <cfg>.
    --clean-first  = Build target 'clean' first, then build.
                     (To clean only, use --target 'clean'.)
    --use-stderr   = Ignored.  Behavior is default in CMake >= 3.0.
    --             = Pass remaining options to the native tool.
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 1

  build script failed, must exit now', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
`

What is the problem?

**Comment by huitseeker** [comment]
The Narwhal build involves the build of some protobuf files. This involves having the `protoc` compiler installed, and this compiler is written in C.
The rust build is trying here, through a dependency, to bring in a version of the protoc source code and compile it, but fails to find:
- a version of `zlib` (not a fatal issue),
- a recent version of `cmake` (blocker).

How to install those depends on your exact distribution and the contents of your VM. /cc @Clay-Mysten / @randall-Mysten for the potential doc update.

**Reply by randall-Mysten** [reply]
do we have an idea of how recent, or a specific version min requirement for this?

**Reply by huitseeker** [reply]
I believe we have tested successful build with a version of cmake as old as the one shipped in Ubuntu bionic (cmake 3.10.2), but ultimately this is only rigorously solvable by looking into the protobuf compiler's installation instructions.

**Reply by sochandmade** [reply]
i has problem too,suggest solve

**Reply by Clay-Mysten** [reply]
> i has problem too,suggest solve

Does upgrading your packages resolve this?

**Comment by Clay-Mysten** [comment]
When you say official doc, do you mean?
https://docs.sui.io/devnet/build/install

And what instructions are you using to build? Sui or Narwhal? Please include the link.

Finally, ensure all of your packages are up to date and include details about your environment. We do call for `cmake` in the Install doc above:
https://docs.sui.io/devnet/build/install#cmake

We do not call for `zlib` though. @huitseeker, should we? For Sui in general? Thanks for the feedback.

**Reply by huitseeker** [reply]
> We do not call for zlib though. @huitseeker, should we? For Sui in general? 

No, I believe zlib is an optional dependency of protoc.

**Reply by Clay-Mysten** [reply]
> > We do not call for zlib though. @huitseeker, should we? For Sui in general?
> 
> No, I believe zlib is an optional dependency of protoc.

Thanks, Francois. @har1s1m, please update `cmake` per:
https://riptutorial.com/cmake/example/4459/cmake-installation

And let us know if you still experience issues.

**Comment by sochandmade** [comment]
i resolve this problem,fullnode sui is start, thank

Vào Th 6, 12 thg 8, 2022 vào lúc 01:19 Clay-Mysten <
***@***.***> đã viết:
